packages/appsync-modelgen-plugin/src/utils/process-belongs-to.ts

```diff
@@ -58,3 +53,35 @@ export function processBelongsToConnection(
     targetName: connectionFields[0] || makeConnectionAttributeName(model.name, field.name),
   };
 }
+
+export function getBelongsToConnectedField(field: CodeGenField, model: CodeGenModel, connectedModel: CodeGenModel, connectionInfo: CodeGenDirective): CodeGenField | undefined {
+  if(connectionInfo.arguments.fields) {
+    let indexDirective = flattenFieldDirectives(model).find(dir => {
+      return dir.name === 'index' && dir.fieldName === connectionInfo.arguments.fields[0];
+    });
+
+    if(indexDirective) {
+      let theIndex = indexDirective;
+      let otherSideConnected = flattenFieldDirectives(connectedModel).find(dir => {
+        return (dir.name === 'hasOne' || dir.name === 'hasMany') && dir?.arguments?.indexName === theIndex.arguments.name;
```

Comment:
Does `hasOne` directive has the `indexName` defined? What is the case for the `hasOne` with `indexName` here?

Reply:
That shouldn't have been there - this was written purposely in response to a specific case, I made a change now to generalize it